src/solid/index.ts

```diff
@@ -0,0 +1,37 @@
+import { createEffect } from "solid-js"
+import autoAnimateBase, {
+  AutoAnimateOptions,
+  AutoAnimationPlugin,
+} from "../index"
+
+/**
+ * This code is taken from https://github.com/lxsmnsyc/solid-auto-animate/blob/main/packages/solid-auto-animate/src/index.ts
+ */
+
+declare module "solid-js" {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace JSX {
+    interface Directives {
+      autoAnimate: Partial<AutoAnimateOptions> | AutoAnimationPlugin | true
+    }
+  }
+}
+
+export function autoAnimate<T extends HTMLElement>(
+  element: T,
+  options: () => Partial<AutoAnimateOptions> | AutoAnimationPlugin | true
+): void {
+  createEffect(() => {
+    const currentOptions = options()
+    autoAnimateBase(element, currentOptions === true ? {} : currentOptions)
+  })
+}
+
+export function createAutoAnimate<T extends HTMLElement>(
```

Comment:
I called this createAutoAnimate instead of useAutoAnimate because it does not need a context / use use*** hook inside

Reply:
take note that the naming scheme is *grammatical* not necessarily whether or not it relies on context or other hidden stuff. `use`/`create` doesn't really matter here.